src/umgmt/user.c

```diff
@@ -538,6 +547,119 @@ long int um_user_get_flags(const um_user_t *user)
     return user->shadow.flags;
 }
 
+/**
+ * Check if an user has any running processes / check if user is logged in
+ *
+ * @param user User to use.
+ * @param running Set to true if user has running processes
+ * @return Error code - 0 on success.
+ *
+ */
+int um_user_has_running_proc(const um_user_t *user, bool *running)
+{
+    return user_processes(user, running, false);
+}
+
+/**
+ * Kill all of user's processes
+ *
+ * @param user User to use.
+ *
+ * @return Error code - 0 on success.
+ *
+ */
+int um_user_kill_all_proc(const um_user_t *user)
+{
+    // set but unchanged, required because of 'user_processes()' function signature
+    bool *running;
```

Comment:
This should be a `bool running;` and `user_processes()` should recieve `&running` as its parameter. Otherwise, functionality of `user_processes()` will not work due to setting of the `has_running` variable inside of the function.